### PR TITLE
Add script to list unlabeled PR's since latest release

### PR DESCRIPTION
**Pull request by rht**
This is an automation of step 1 in https://github.com/projectmesa/mesa/pull/1917#issuecomment-1871352058. Step 2-6 shouldn't be automated, because step 5 is manual, and is tied to these steps. In order for step 2-6 to be fully automated, step 7 needs to happen first (which is doable, actually).

@EwoutH given that the file is now CONTRIBUTING.md instead of .rst, you may add the release checklist there.

### CODE REVIEW

**Comment by rht** [comment]
See https://github.com/orgs/community/discussions/10038.

**Comment by EwoutH** [comment]
Seeing unlabeled merged PRs is as easy as [this](https://github.com/projectmesa/mesa/pulls?q=is%3Apr+is%3Amerged+no%3Alabel+), right? Anything since december 2023 should be labeled.

**Comment by rht** [comment]
That's true. Is there a way to filter to cut off further the earliest date? But if a bulk of the checklist can be automated, it would be easier to read the info from the command line.

**Comment by EwoutH** [comment]
Yes, see [the docs](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-created-or-last-updated).

**Comment by EwoutH** [comment]
@rht do you mind closing this PR? I think using the GitHub URL is automated enough for now.

**Comment by rht** [comment]
As I said on https://github.com/projectmesa/mesa/pull/2047#issuecomment-1968029387, once again, if almost everything in the checklist is automated, it is better off to use the CLI script. This PR is a first step in that direction. I usually don't draft the releases, but if you, @jackiekazil , and @tpike3 find the automation to be not worth it, you may close this PR.

**Comment by jackiekazil** [comment]
I will do a review before the end of the week. I want to make sure I understand things completely.

**Comment by rht** [comment]
The steps are:
1. (automated) Use the script in this PR to check for unlabeled PRs. Do not allow to proceed until the list is empty.
2. (automated) Automatically generate a release note with https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release, and auto-append this to HISTORY.md.
3. (manual) Add highlights to the release
4. (can be partly scripted) Open a PR that includes the HISTORY.md change and `__init__.py` version bump. Wait until it is merged.
5. (automated) Release using https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release
6. (manual) Check on PyPI if the release is there
7. (automated) Open a PR to append `-dev` to the next minor release version.

As you may notice, it is much simpler than https://github.com/projectmesa/mesa/pull/1917#issuecomment-1871352058, because the generate release note step (without drafting a release) already has an API.

**Comment by EwoutH** [comment]
Just to re-iterate, In my opinion, this script is unnecessary, since you can just go to https://github.com/projectmesa/mesa/pulls?q=is%3Apr+is%3Amerged+no%3Alabel+merged%3A%3E%3D2023-12-01 and see which PRs need a label yet. I also don't know yet how this helps with automation.

However, if this is useful for others I'm not against merging this.

**Review by jackiekazil — APPROVED** [review]

**Comment by jackiekazil** [comment]
I am going to merge based on the discuss. If we don't end up using this in the future we can remove it.

**Pull request merged**
